### ISSUE TITLE
Invalid code generated when variables with reserved keywords as names require defining a local variable for its native representation

### ISSUE THREAD

**Issue by elinor-fung** [dotnet/runtimelab#1161]
Support for using reserved keywords as names for by-value blittable arguments was added with #1105. Using reserved keywords as names still generates invalid code when marshalling requires using a local variable for its native representation:
```C#
[GeneratedDllImport("DoesNotExist")]
public static partial void Method(out int @out);
```

Generated code has invalid names:
```C#
fixed (int* __@out_gen_native = &@out)
    Method__PInvoke__(__@out_gen_native);
```

cc @AaronRobinsonMSFT @jkoritzinsky

**Comment by elinor-fung** [comment]
Totally fine with saying we don't support this if it adds a bunch of complexity. Just figured we should call it out (I hit it in one of the libraries - and promptly just changed the variable name).

**Comment by jkoritzinsky** [comment]
Fixed by #1167
